src/shared/utils/url.ts

```diff
@@ -10,22 +8,22 @@ export function getFilenameFromFilePath(path: string) {
 }
 
 export function getFilePathParts(path: string) {
-  if (path === '') {
+  if (!path || path === '') {
     return []
   }
-  return path?.split('/') ?? []
+  return path.split('/') ?? []
 }
 
 export function getFilenameFromPath(path?: string) {
-  if (!isString(path)) {
+  if (typeof path !== 'string') {
```

Comment:
I noticed we use isString a lot, but that's basically just a wrapper for this call... can probably just remove that bloat elsewhere too